### PR TITLE
Harden computeFrustumPlanes() in FrustumPlanes.ts

### PR DESCRIPTION
**Pull request by markschlosseratbentley**
Fixes https://github.com/iTwin/itwinjs-backlog/issues/670

In the case of the above bug, the front rect of a scaled-down frustum for readPixels() can become rather tiny because the source view frustum was already rather small. Deriving the front plane's normals from a small rect resulted in a near-zero plane normal, causing computeFrustumPlanes() to return an empty array of planes. This caused the rendering system to eventually assert.

To fix the problem, we now derive the front plane from the rear plane.

We also generally hardened the code in computeFrustumPlanes() by ensuring that the normals are only ever calculated from points of the frustum box that yield a 90 degree angle (this was not done previously).  We did this by reordering the point used for the cross product function. This reordering also makes it clearer what the values actually mean.

More documentation has been added to this source file in order to make it easier to debug and read in the future.

Furthermore, a test has been added (FrustumPlanes.test.ts) which exercises the particular bad frustum which prompted this work, verifying that it now processes properly. Some good frustums have been added to that test file as well.

- [x] Image Tests

### CODE REVIEW

**Review by markschlosseratbentley — COMMENTED** [review]
Image Tests are currently running.

File: core/common/src/test/FrustumPlanes.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { expect } from "chai";
+import { FrustumPlanes } from "../geometry/FrustumPlanes";
+import { Frustum } from "../Frustum";
+import { Point3d } from "@itwin/core-geometry";
+
+describe("FrustumPlanes", () => {
```
Comment (pmconne) [inline core/common/src/test/FrustumPlanes.test.ts:10]: Also test how it handles a bad frustum (e.g., input points do not define a valid frustum)?

Comment (markschlosseratbentley) [reply core/common/src/test/FrustumPlanes.test.ts:10]: done

**Review by pmconne — APPROVED** [review]

**Comment by pmconne** [comment]
Backporting to 3.6 and 3.7, right? Issue was reported against 3.6.2.

**Comment by calebmshafer** [comment]
@pmconne 3.7 should be the only one needed as 3.7.0 is out and we'll release a 3.7.1 Monday (if we haven't already).

**Comment by pmconne** [comment]
@calebmshafer you're ok with introducing a regression in 3.6 and not fixing it in 3.6?

**Comment by markschlosseratbentley** [comment]
@calebmshafer just 3.7 backport or 3.6 as well per the above comments?

**Comment by markschlosseratbentley** [comment]
@mergifyio backport release/3.7.x

**Comment by markschlosseratbentley** [comment]
@pmconne @calebmshafer I completed a backport to 3.7. Let me know if we decide on a 3.6 backport. Any backport needs a bit of manual tweaking due to conflicts in FrustumPlanes.ts.

**Pull request merged**